### ISSUE TITLE
Looking for synchronous API

### ISSUE THREAD

**Issue by paulanthonymorris** [dsplaisted/PCLStorage#30]
Now that Universal Windows has support for returning to the System.IO.File, etc. API's with their synchronous method calls, we would like to see support in PCLStorage for synchronous calls. We prefer to retain the use of synchronous file IO patterns in our Android and iOS code and update our asynchronous Windows IO patterns to match that (rather than introduce the complexity of asynchronous file IO patterns into our Android and iOS code). What is your opinion? Are there plans to support this in PCLStorage?

**Comment by dsplaisted** [comment]
I think that now that System.IO file APIs are available everywhere, there's not much of a need for PCL Storage.  Is there a reason not to just use those APIs directly?

**Comment by paulanthonymorris** [comment]
Thanks for the quick response. So System.IO.File is available on all the platforms but I can’t access System.IO.File calls from within a portable lib so as I understand it I need some abstraction down there (like System.IO.Stream). Am I missing something?

**Comment by dsplaisted** [comment]
Yes, a lot of the new ".NET Core" APIs are not currently available in PCLs targeting Xamarin.  They should be eventually.  So for now you would probably want to create your own abstraction if you want to use the APIs from a PCL.

**Comment by paulanthonymorris** [comment]
Would you consider it admissible to add synchronous implementations to the PCLStorage API?

**Comment by dsplaisted** [comment]
I would prefer not to add synchronous APIs to the PCLStorage package itself.  They will be obsolete once the System.IO APIs are available in PCLs.

If you want a package with synchronous APIs in the meantime, I'd welcome a PR to create a separate package for that.

**Comment by paulanthonymorris** [comment]
Ok then. We may just write a thin abstraction around the File and Directory API's to tide us over. Thanks for your input.
